x-pack/plugins/apm/server/lib/helpers/get_document_sources.ts

```diff
@@ -111,21 +175,29 @@ export async function getDocumentSources({
       rollupInterval,
       hasDataBefore,
       hasDataAfter,
+      checkSummaryExistsField: source.meta.checkSummaryExistsField,
     };
   });
 
-  const hasAnyDataBefore = checkedSources.some(
-    (source) => source.hasDataBefore
-  );
+  const hasAnyDataBefore = checkedSources
+    .filter((source) => !source.checkSummaryExistsField)
+    .some((source) => source.hasDataBefore);
```

Comment:
I don't understand this change. What's the logic here?

Reply:
there are 2 things here
1. Why we need `hasAnyDataBefore`
Answer - If a client freshly started using our product, which means they will not have any old data to compare with. Hence we need this value to calculate `hasDocs` and `hasDurationSummary` later in the logic.
2. Why this code change where i am filtering based on `checkSummaryExistsField` ?
Answer - We were firing a combined `msearch` query to check for `hasDocs` only previously. Now we have added more queries to this single `msearch` query to also check for `duration summary field`. Now since the `checkedSource` has both queries for `hasDocs` and `hasDurationSummary` field, we need to filter out the results for `hasDocs` only

Reply:
@achyutjhunjhunwala why do have to exclude `hasDurationSummary` searches?

Reply:
Will include it now 👍🏼